### PR TITLE
Networking API

### PR DESCRIPTION
**Pull request by CoolMineman**
Adds a simple networking api!

### CODE REVIEW

**Comment by CoolMineman** [comment]
5 min till I just merge

File: src/main/java/io/github/minecraftcursedlegacy/accessor/AccessorAbstractPacket.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package io.github.minecraftcursedlegacy.accessor;
+
+import org.spongepowered.asm.mixin.Mixin;
```
Comment (Ueaj-Kerman) [inline src/main/java/io/github/minecraftcursedlegacy/accessor/AccessorAbstractPacket.java:3]: no

File: src/main/java/io/github/minecraftcursedlegacy/accessor/AccessorAbstractPacket.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import org.spongepowered.asm.mixin.gen.Invoker;
+
+import net.minecraft.packet.AbstractPacket;
+
+@Mixin(AbstractPacket.class)
+public interface AccessorAbstractPacket {
+	@Invoker("register")
+	static void register(int var0, boolean flag, boolean flag1, Class arg) {
```
Comment (Ueaj-Kerman) [inline src/main/java/io/github/minecraftcursedlegacy/accessor/AccessorAbstractPacket.java:11]: no

File: src/main/java/io/github/minecraftcursedlegacy/api/networking/PluginChannel.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package io.github.minecraftcursedlegacy.api.networking;
+
+import io.github.minecraftcursedlegacy.impl.networking.PluginMessagePacket;
+import net.minecraft.client.Minecraft;
+import net.minecraft.entity.player.ServerPlayer;
+import net.minecraft.network.PacketHandler;
+
+public abstract class PluginChannel {
```
Comment (Ueaj-Kerman) [inline src/main/java/io/github/minecraftcursedlegacy/api/networking/PluginChannel.java:8]: bad name

File: src/main/java/io/github/minecraftcursedlegacy/api/networking/PluginChannel.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	/**
+	 * Returns the unique channel identifier of this channel in the form of modid:channelname
+	 * This should always have the same return value.
+	 * @return Channel Identifier
+	 */
+	public abstract String getChannelIdentifier();
```
Comment (Ueaj-Kerman) [inline src/main/java/io/github/minecraftcursedlegacy/api/networking/PluginChannel.java:14]: use a namespaced id here, string bad

File: src/main/java/io/github/minecraftcursedlegacy/api/networking/PluginChannelRegistry.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package io.github.minecraftcursedlegacy.api.networking;
+
+import io.github.minecraftcursedlegacy.impl.networking.PluginChannelRegistryImpl;
+
+public class PluginChannelRegistry {
```
Comment (Ueaj-Kerman) [inline src/main/java/io/github/minecraftcursedlegacy/api/networking/PluginChannelRegistry.java:5]: make this a final interface

File: src/main/resources/fabric.mod.json [hunk 6]
```diff
@@ -20,7 +20,8 @@
   "entrypoints": {
     "init": [
       "io.github.minecraftcursedlegacy.impl.levelgen.LevelGenImpl",
-      "io.github.minecraftcursedlegacy.impl.registry.RegistryImpl"
+      "io.github.minecraftcursedlegacy.impl.registry.RegistryImpl",
+      "io.github.minecraftcursedlegacy.impl.networking.NetworkingImpl"
```
Comment (Ueaj-Kerman) [inline src/main/resources/fabric.mod.json:24]: reduce this to 1 entryoinaet

File: src/main/resources/api.accessors.json [hunk 7]
```diff
@@ -8,7 +8,8 @@
     "AccessorRecipeRegistry",
     "AccessorShapedRecipe",
     "AccessorShapelessRecipe",
-    "AccessorTileItem"
+    "AccessorTileItem",
```
Comment (Ueaj-Kerman) [inline src/main/resources/api.accessors.json:11]: TileItemAccess, not AccessorTileItem this isn't mcp

File: src/main/java/io/github/minecraftcursedlegacy/mixin/MixinPendingConnection.java [hunk 8]
```diff
@@ -5,7 +5,7 @@
 import org.spongepowered.asm.mixin.injection.Inject;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 
-import io.github.minecraftcursedlegacy.impl.vanillachecker.VanillaCheckerImpl;
+import io.github.minecraftcursedlegacy.impl.server.VanillaCheckerImpl;
```
Comment (Ueaj-Kerman) [inline src/main/java/io/github/minecraftcursedlegacy/mixin/MixinPendingConnection.java:8]: only good change I've seen so far

File: src/main/java/io/github/minecraftcursedlegacy/impl/networking/PluginMessagePacket.java [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+import net.minecraft.network.PacketHandler;
+import net.minecraft.packet.AbstractPacket;
+
+//Similar to Plugin Message but supports 
+public class PluginMessagePacket extends AbstractPacket {
+	public String channel;
```
Comment (Ueaj-Kerman) [inline src/main/java/io/github/minecraftcursedlegacy/impl/networking/PluginMessagePacket.java:12]: no, namespaced id please.

File: src/main/java/io/github/minecraftcursedlegacy/impl/networking/NetworkingImpl.java [hunk 10]
```diff
@@ -0,0 +1,12 @@
+package io.github.minecraftcursedlegacy.impl.networking;
+
+import io.github.minecraftcursedlegacy.accessor.AccessorAbstractPacket;
+import net.fabricmc.api.ModInitializer;
+
+public class NetworkingImpl implements ModInitializer {
+
+	@Override
+	public void onInitialize() {
+		AccessorAbstractPacket.register(250, true, true, PluginMessagePacket.class);
```
Comment (Ueaj-Kerman) [inline src/main/java/io/github/minecraftcursedlegacy/impl/networking/NetworkingImpl.java:10]: nice magic numbers and test code

File: src/main/java/io/github/minecraftcursedlegacy/api/networking/PluginChannel.java [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+	 */
+	public abstract void onRecieve(PacketHandler arg, byte[] data);
+	
+	/**
+	 * Server packet send method
+	 * @param data The data to send
+	 * @param player The player to send the data to
+	 */
+	public void send(byte[] data, ServerPlayer player) {
+		player.field_255.method_835(new PluginMessagePacket(getChannelIdentifier(), data));
```
Comment (Ueaj-Kerman) [inline src/main/java/io/github/minecraftcursedlegacy/api/networking/PluginChannel.java:29]: deobf this field and method, or leave a comment as to what it does

**Review by Ueaj-Kerman — COMMENTED** [review]
terrible pls revert

File: src/main/java/io/github/minecraftcursedlegacy/api/networking/PluginChannel.java (hunk 4 above)
Comment (valoeghese) [inline src/main/java/io/github/minecraftcursedlegacy/api/networking/PluginChannel.java:14]: use namespaced id

File: src/main/java/io/github/minecraftcursedlegacy/api/networking/PluginChannel.java [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+public abstract class PluginChannel {
+	/**
+	 * Returns the unique channel identifier of this channel in the form of modid:channelname
+	 * This should always have the same return value.
+	 * @return Channel Identifier
+	 */
+	public abstract String getChannelIdentifier();
+	
+	/**
+	 * Called when this channel recieves a packet
```
Comment (valoeghese) [inline src/main/java/io/github/minecraftcursedlegacy/api/networking/PluginChannel.java:17]: full stop at the end of javadoc sentences

File: src/main/java/io/github/minecraftcursedlegacy/impl/networking/PluginMessagePacket.java (hunk 9 above)
Comment (valoeghese) [inline src/main/java/io/github/minecraftcursedlegacy/impl/networking/PluginMessagePacket.java:12]: namespaced id

File: src/main/java/io/github/minecraftcursedlegacy/impl/networking/PluginMessagePacket.java [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+	public byte[] data;
```
Comment (valoeghese) [inline src/main/java/io/github/minecraftcursedlegacy/impl/networking/PluginMessagePacket.java:13]: why public

**Review by valoeghese — COMMENTED** [review]
Only a few problems. Should have used NamespacedId

File: src/main/java/io/github/minecraftcursedlegacy/api/networking/PluginChannel.java (hunk 12 above)
Comment (Ueaj-Kerman) [reply src/main/java/io/github/minecraftcursedlegacy/api/networking/PluginChannel.java:17]: no that's lame

**Pull request merged**